CHANGELOG.md

```diff
@@ -16,17 +16,17 @@ Here you can find upgrade changes in between releases and upgrade instructions.
   known to be needed, with the exception of the user pods outgoing
   communication, where all outgoing communication is allowed by default.
   
-  Note that these network policies only influence network communication in a
+  Note that these NetworkPolicies only influence network communication in a
   Kubernetes cluster if a NetworkPolicy controller enforce them, such as Calico.
 
-  With network policies enabled, you may require additional configuration,
-  especially for deployments that include additional components that access
-  JupyterHub pods directly (i.e. not through the `proxy-public` service).
-
   See the [security
   documentation](https://zero-to-jupyterhub.readthedocs.io/en/latest/administrator/security.html#kubernetes-network-policies)
   for more details on this.
 
+- The Helm chart configuration `proxy.networkPolicy` has been removed,
```

Comment:
This change is my biggest concern here. I don't love it because it seems to be baking too much of the implementation into the configuration. When we finally manage to land the traefik proxy implementation (#1162), this configuration structure will change again when nothing relevant to *users* should really need. What do *users* need to know about the separation of autohttps and chp that we couldn't represent without separating them?

Reply:
We will still have a dedicated proxy for TLS termination i think, given that traefik doesnt support HA with lets encrypt.

This helm chart has a separate pod for TLS termination atm, and combining the config of two pods into one has caused it to be confusion in general. I'm think im for doing this even if it was a standalone refactoring without any other purpose to reduce that confusion.

If you feel strongly about it, i can revert it back, locating chp netpol config under proxy.netpol again.

Reply:
To me, this change seems like it would *increase* confusion to have two network policies applying to what users should see as one resource (and indeed will be one resource if we manage to land the traefik proxy). I am thinking of the example of

> Q: I set proxy.chp.networkPolicy, why is it not having any effect?
> A: that's because you have enabled autohttps, which means that proxy.traefik.networkPolicy governs proxy-public, not proxy.chp.networkPolicy. If you change how you configure https, your network policy will do what you expect, or you need to move the configuration to proxy.traefik.networkPolicy

Or is that not an accurate description of the behavior here?